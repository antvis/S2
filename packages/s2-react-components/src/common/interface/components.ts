export interface BaseComponentProps<T> {
  /**
   * 标题
   * @default "文字对齐"
   */
  title?: React.ReactNode;

  /**
   * 默认是否折叠
   * @default false
   */
  defaultCollapsed?: boolean;

  /**
   * 默认配置
   */
  defaultOptions?: Partial<T>;

  /**
   * 子节点
   */
  children?: React.ReactNode;
}
