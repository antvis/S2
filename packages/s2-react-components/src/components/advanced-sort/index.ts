export { AdvancedSort } from './advanced-sort';
export { CustomSort } from './custom-sort';

export type {
  AdvancedSortBaseProps,
  AdvancedSortProps,
  CustomSortProps,
  Dimension,
  RuleItem,
  RuleOption,
  RuleValue,
} from './interface';
