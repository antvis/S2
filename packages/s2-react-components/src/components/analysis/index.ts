export * from './advanced-sort';
export * from './drill-down';
export * from './export';
export * from './switcher';
