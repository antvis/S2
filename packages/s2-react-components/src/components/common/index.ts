export { ResetButton } from './reset-button';
export { ResetGroup } from './reset-group';
