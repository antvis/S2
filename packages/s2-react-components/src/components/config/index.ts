export {
  FrozenInputNumber,
  FrozenPanel,
  type FrozenPanelOptions,
  type FrozenPanelProps,
} from './frozen-panel';

export {
  TextAlignPanel,
  type TextAlignPanelOptions,
  type TextAlignPanelProps,
} from './text-align-panel';

export {
  ColorBox,
  ColorPickerPanel,
  ThemePanel,
  type ThemePanelOptions,
  type ThemePanelProps,
} from './theme-panel';
