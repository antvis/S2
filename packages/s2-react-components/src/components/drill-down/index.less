@import '@antv/s2/esm/shared/styles/drill-down.less';
