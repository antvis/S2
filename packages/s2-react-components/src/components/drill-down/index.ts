export { DrillDown } from './drill-down';

export type { DrillDownDataSet, DrillDownProps } from './interface';
