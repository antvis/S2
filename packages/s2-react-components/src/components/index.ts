export * from './common';

export * from './advanced-sort';
export * from './drill-down';
export * from './export';
export * from './switcher';

export * from './frozen-panel';
export * from './text-align-panel';
export * from './theme-panel';
