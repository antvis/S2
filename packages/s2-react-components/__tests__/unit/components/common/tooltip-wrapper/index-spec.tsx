import { TooltipWrapper } from '@/components';
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';

describe('Tooltip Wrapper Component Tests', () => {
  test('should render correctly', () => {
    const { asFragment } = render(
      <TooltipWrapper title="测试" className="test" />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('should render custom content', () => {
    render(
      <TooltipWrapper title="测试">
        <div>自定义内容</div>
      </TooltipWrapper>,
    );

    expect(screen.getByText('自定义内容')).toBeTruthy();
  });

  test('should show custom label when tooltip clicked', () => {
    render(
      <TooltipWrapper title="测试" trigger="click">
        <div>自定义内容</div>
      </TooltipWrapper>,
    );

    fireEvent.click(screen.getByText('自定义内容'));

    expect(screen.getByText('测试')).toBeTruthy();
  });
});
