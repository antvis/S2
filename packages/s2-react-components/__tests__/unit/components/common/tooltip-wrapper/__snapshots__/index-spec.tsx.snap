// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Button Component Tests should render correctly 1`] = `
<DocumentFragment>
  <span
    class="antv-s2-tooltip-wrapper test"
  />
</DocumentFragment>
`;
